### PR TITLE
Documentation

### PR DESCRIPTION
**Pull request by afmagee42**
This PR adds `mkdocs` documentation, namely:
- A brief overview of how `cladecombiner` sees the world (some potential tip taxon gotchas and the difference between taxonomy and nomenclature)
- An API reference
- An introduction to how to use cladecombiner to do `BasicPhylogeneticAggregation`

Running `mkdocs` with `numpy` docstring formatting found several errors in the API documentation, which have been fixed.

### CODE REVIEW

File: docs/fixed_agg_workflow.md [hunk 1]
```diff
@@ -0,0 +1,196 @@
...
+Aggregation requires us to be explicit about whether we are truly aggregating or not.
+Three of these represent actual aggregations of the observed taxa, as we have observed lineages which are children of BA.2, BA.5, and JN.
+However, as we have no children of KP.1.1 in our observed lineages, it is a tip.
+
+```
+target_taxa = [
+    cladecombiner.Taxon("BA.2", False),
+    cladecombiner.Taxon("BA.2.86.1", False),
+    cladecombiner.Taxon("BA.5", False),
+    cladecombiner.Taxon("KP.1.1", True),
```
Comment (swo) [inline docs/fixed_agg_workflow.md:110]: Is this equivalent to saying "map KP.1.1 to itself, and not to anything above it"?

File: docs/fixed_agg_workflow.md [hunk 2]
```diff
@@ -0,0 +1,196 @@
...
+Taxon(JN.6, tip=True)      : Taxon(BA.2.86.1, tip=False)
+Taxon(XCU, tip=True)       : Taxon(BA.2, tip=False)
+Taxon(BA.2.86, tip=True)   : Taxon(BA.2, tip=False)
+Taxon(FW.1.1.1, tip=True)  : Taxon(BA.2, tip=False)
+Taxon(BA.2, tip=True)      : Taxon(BA.2, tip=False)
+```
+
+## Using and saving the aggregation
+
+The resulting mapping can be used to re-name taxa in, for example, line-list data.
```
Comment (swo) [inline docs/fixed_agg_workflow.md:143]:
```suggestion
The resulting mapping can be used to rename taxa in, for example, line list data.
```

File: docs/fixed_agg_workflow.md [hunk 3]
```diff
@@ -0,0 +1,196 @@
...
+A `dict[str,str]` can be obtained via `res.to_str()` which can be fed into the `replace` methods in [pandas](https://pandas.pydata.org/pandas-docs/stable/reference/api/pandas.DataFrame.replace.html) or [polars](https://docs.pola.rs/api/python/stable/reference/expressions/api/polars.Expr.replace.html).
+
+The `res.to_str()` dictionary can be saved for posterity by writing to file using python's [`json.dumps()`](https://docs.python.org/3/library/json.html) function.
+
+Saving the python script for posterity is also recommended.
```
Comment (swo) [inline docs/fixed_agg_workflow.md:148]: On the call today we talked a little bit about pickling. The trick with pickling anything other than basic Python objects is that you can run into versioning trouble, IIRC. A problem for another time.

File: docs/fixed_agg_workflow.md [hunk 4]
```diff
@@ -0,0 +1,196 @@
...
+Where the saved map is an exact record of how the particular input taxa were mapped, the script can be used on a new set of input taxa to produce a new mapping.
+
+## Important settings
+
+When creating the `BasicPhylogeneticAggregator`, there are two important arguments: `off_target` and `sort_clades`.
+
+### `off_target`
```
Comment (swo) [inline docs/fixed_agg_workflow.md:155]: To check my understanding, if a taxon is "off target," that means that it doesn't have an ancestor taxon in the list of "target" taxa?

File: docs/index.md [hunk 5]
```diff
@@ -0,0 +1,119 @@
+# cladecombiner
+
+This package provides functionality for working with taxa, with a focus on phylogenetic aggregation to higher-order taxa.
+
+While the functionality is intended to be general, current out-of-the-box support is limited to Pango lineages for SARS-CoV-2.
+
+## Taxon, clade, or lineage?
+
+For the purposes of cladecombiner and its documentation, a taxon is a unit of taxonomy which includes both [Nextstrain clades](https://nextstrain.org/blog/2021-01-06-updated-SARS-CoV-2-clade-naming) and [Pango lineages](https://www.nature.com/articles/s41564-020-0770-5) for SARS-CoV-2.
```
Comment (swo) [inline docs/index.md:9]:
```suggestion
For the purposes of cladecombiner and its documentation, a taxon is a unit of taxonomy. [Nextstrain clades](https://nextstrain.org/blog/2021-01-06-updated-SARS-CoV-2-clade-naming) and [Pango lineages](https://www.nature.com/articles/s41564-020-0770-5) for SARS-CoV-2 are both examples of taxa.
```

File: docs/index.md [hunk 6]
```diff
@@ -0,0 +1,119 @@
...
+We reserve clade and lineage (unless referring to Nextstrain clades or Pango lineages) for the usual sense of monophyletic groupings defined by an ancestor and all its descendants.
```
Comment (swo) [inline docs/index.md:10]:
```suggestion
We reserve the terms "clade" and "lineage" (unless referring to Nextstrain clades or Pango lineages) for the usual sense of monophyletic groupings defined by an ancestor and all its descendants.
```

File: docs/index.md [hunk 7]
```diff
@@ -0,0 +1,119 @@
...
+
+As clades can contain clades and lineages can contain lineages, taxa can contain taxa.
+Cladecombiner assumes that there is some base level of taxonomy which can be represented as a phylogeny with every taxon at this level as a tip.
+Cladecombiner defaults to monophyletic groupings of these taxa where possible, but allows for non-monophyletic aggregated taxa.
+
+### Ancestral versus tip taxa
+
+When tracking ongoing evolution, the naming of new monophyletic taxa creates potential ambiguity when using the name of the now-non-monophyletic ancestor.
```
Comment (swo) [inline docs/index.md:18]:
```suggestion
When tracking ongoing evolution, the naming of new monophyletic taxa creates potential ambiguity when using the name of the now non-monophyletic ancestor.
```

File: docs/index.md [hunk 8]
```diff
@@ -0,0 +1,119 @@
...
+For example, JN.1 can refer to the ancestral taxon, which has children such as KP.2.
+It can also refer to JN.1 the taxon we can observe in data (a sequence which belongs to JN.1 but not any more specifically named descendant).
+
+Cladecombiner distinguishes between these two senses in which a taxon name can be used by adding tips to the underlying taxonomy tree to represent the non-monophyletic remainder.
+Thus, if needed, the ancestral `Taxon("JN.1", is_tip=False)` can have as a descendant tip `Taxon("JN.1", is_tip=True)`.
+
+## Usage
+
+If we have line-list data like this:
```
Comment (swo) [inline docs/index.md:27]:
```suggestion
If we have line list data like this:
```

**Review by swo — APPROVED** [review]
Awesome!

Some minor copy edits and some questions that are more for my education than anything else.

See #29 for a suggestion about CI for mkdocs.

File: docs/fixed_agg_workflow.md (hunk 1 above)
Comment (afmagee42) [reply docs/fixed_agg_workflow.md:110]: Yes. Adding that to the text above for clarity, and cross-linking to the section we describe _why_ we have to do this.

File: docs/fixed_agg_workflow.md (hunk 4 above)
Comment (afmagee42) [reply docs/fixed_agg_workflow.md:155]: Oh, good catch, I had a really bad description there. Yes, "off-target" means "no ancestor present in specified targets"

**Pull request merged**